platform/ui/src/components/Input/Input.jsx

```diff
@@ -1,50 +1,47 @@
 import React from 'react';
+import PropTypes from 'prop-types';
 import Label from '../Label';
+import classnames from 'classnames';
+
+const baseInputClasses =
+  'shadow transition duration-300 appearance-none border rounded w-full py-2 px-3 text-sm text-gray-700 hover:border-gray-500 leading-tight focus:border-gray-500 focus:outline-none';
+
+const transparentClasses = {
+  true: 'bg-transparent',
+  false: '',
+};
 
 const Input = ({
   label,
   containerClassName = '',
   labelClassName = '',
   className = '',
-  transparent = defaults.transparent,
-  ...rest
+  transparent = true,
+  id,
+  ...otherProps
 }) => {
-  const getClasses = () => {
-    const classes = [];
-
-    classes.push(transparentClasses[transparent]);
-
-    return classes.join(' ');
-  };
-
-  const input = (
-    <input
-      className={`shadow transition duration-300 appearance-none border rounded w-full py-2 px-3 text-sm text-gray-700 hover:border-gray-500 leading-tight focus:border-gray-500 focus:outline-none ${getClasses()} ${className}`}
-      id="id"
-      {...rest}
-    />
+  return (
+    <div className={classnames('flex flex-col flex-1', containerClassName)}>
+      <Label className={labelClassName} text={label} for={id ? id : ''}></Label>
+      <input
+        className={classnames(
+          className,
+          baseInputClasses,
+          transparentClasses[transparent]
+        )}
+        id={id ? id : ''}
```

Comment:
If `id` defaults to empty string, set that as the default in `defaultProps`? If `id` has to be unique, should we auto-generate a string guid?

Reply:
Removed the change for now.